File: app/src/main/java/ru/roscha_akademii/medialib/video/model/local/VideoTable.java
package ru.roscha_akademii.medialib.video.model.local;

import ru.roscha_akademii.medialib.common.CreateTableQueryBuilder;

public class VideoTable {
    public static final String TABLE_NAME = "video";

    public static final String ID = "_id";
    public static final String TITLE = "title";
    public static final String DESCRIPTION = "description";
    public static final String PICTURE_URL = "picture_url";
    public static final String VIDEO_URL = "video_url";

    public static String createTable() {
        return new CreateTableQueryBuilder(VideoTable.TABLE_NAME)
                .index(VideoTable.ID)
                .column(VideoTable.TITLE, CreateTableQueryBuilder.SqlType.STRING)
                .column(VideoTable.DESCRIPTION, CreateTableQueryBuilder.SqlType.STRING)
                .column(VideoTable.PICTURE_URL, CreateTableQueryBuilder.SqlType.STRING)
                .column(VideoTable.VIDEO_URL, CreateTableQueryBuilder.SqlType.STRING)
                .build();
    }
}

File: app/src/main/java/ru/roscha_akademii/medialib/video/model/local/VideoTable.kt
package ru.roscha_akademii.medialib.video.model.local

import ru.roscha_akademii.medialib.common.CreateTableQueryBuilder
import ru.roscha_akademii.medialib.common.CreateTableQueryBuilder.SqlType.STRING


object VideoTable {
    val VERSION = 6

    val TABLE_NAME = "video"

    val ID = "_id"
    val TITLE = "title"
    val DESCRIPTION = "description"
    val PICTURE_URL = "picture_url"
    val VIDEO_URL = "video_url"
    val DURATION = "duration"
    val DATE = "date"

    fun createTable(): String {
        return CreateTableQueryBuilder(TABLE_NAME)
                .index(ID)
                .column(TITLE, STRING)
                .column(DESCRIPTION, STRING)
                .column(PICTURE_URL, STRING)
                .column(VIDEO_URL, STRING)
                .column(DURATION, STRING)
                .column(DATE, STRING)
                .build()
    }
}
